CMakeLists.txt

```diff
@@ -179,11 +179,11 @@ endif()
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   if (APPLE)
     # Detect OS X version. Use '/usr/bin/sw_vers -productVersion' to
-    # extract V from '10.V.x'.
+    # extract V from '11.V.x'.
     exec_program(/usr/bin/sw_vers ARGS
         -productVersion OUTPUT_VARIABLE MACOSX_VERSION_RAW)
     string(REGEX REPLACE
-        "10\\.([0-9]+).*" "\\1"
+        "11\\.([0-9]+).*" "\\1"
         MACOSX_VERSION
```

Comment:
Are you sure this modification will work as you change the version to 10 now?

Reply:
It is working as for now.
I was trying to fix the following.
ld: warning: dylib (/Users/runner/work/1/s/build/lib/libmlpack.dylib) was built for newer macOS version (11.6) than being linked (10.14)
ld: warning: dylib (/usr/local/lib/libarmadillo.dylib) was built for newer macOS version (11.0) than being linked (10.14)
But its not fixed rn.

Reply:
I think I overlooked the details of these lines of code.
But why is there \\ after 11?
And better regex shouldn't be 11\\.([0-9]+)((.[0-9]+)*)
As version can be of form 11.V only.
I will change 11 to 10 back.

Reply:
([0-9]+)(?:.([0-9]+).*)*